src/commonMain/kotlin/com/charleskorn/kaml/YamlNamingStrategy.kt

```diff
@@ -0,0 +1,97 @@
+/*
+
+   Copyright 2018-2021 Charles Korn.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+*/
+
+package com.charleskorn.kaml
+
+public fun interface YamlNamingStrategy {
+    public fun serialNameForYaml(serialName: String): String
+
+    public companion object Builtins {
+        /**
+         * A [YamlNamingStrategy] that converts property names to snake_case (lowercase words separated by underscores).
+         */
+        public val SnakeCase: YamlNamingStrategy = object : YamlNamingStrategy {
+            override fun serialNameForYaml(serialName: String): String = serialName.toDelimitedCase('_')
+
+            override fun toString(): String = "com.charleskorn.kaml.YamlNamingStrategy.SnakeCase"
```

Comment:
Are the `toString()` overrides required?

Reply:
Not necessarily - KotlinX Serialization did that for their Json equivalent of this class, so I did too